### ISSUE TITLE
[MNG-5374] Fix transfer listener after the JSR330 merge

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#6553]
**[Jason van Zyl](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jvanzyl)** opened **[MNG-5374](https://issues.apache.org/jira/browse/MNG-5374?redirect=false)** and commented

The transfer listener is using the logger which is producing ugly output on the screen, and so should be restored to its previous behaviour on the console.

---
No further details from [MNG-5374](https://issues.apache.org/jira/browse/MNG-5374?redirect=false)

**Comment by jira-importer** [comment]
**[Olivier Lamy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=olamy)** commented

a possible fix using log4j2: https://github.com/olamy/maven-3/tree/log4j2

**Comment by jira-importer** [comment]
**[Jason van Zyl](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jvanzyl)** commented

I don't believe pulling in a logging framework like Logback for Log4J is necessary. I have done the following for the following cases:

For the following cases I have done the following:

1) Logging to the console: I simply restored the class that was there such that when logging to the console it uses System.out as it did and the download progress appears as it did.

2) Logging in batch mode: the batch mode transfer listener uses an SLF4J logger and the batch mode transfer listener doesn't report download progress so there is no issue. Download progress would just create a bunch of noise. The size and the speed at which it is downloaded are logged.

3) Logging to a file: same as 2) except it's all diverted to the specified file.

4) I created two protected methods in MavenCli so that integrators can supply their own console and batch transfer listeners if they wish.

I think that covers everything and avoids having to bring in Logback or Log4J. I think I agree with Mark's notion of possibly making this implementation our own, but taking SLF4J Simple and just changing the package space as never to interfere.
